app/scripts/services/events.js

```diff
@@ -12,31 +12,30 @@ angular.module('openshiftConsole')
 
       var EVENTS_TO_SHOW_BY_REASON = _.get(window, 'OPENSHIFT_CONSTANTS.EVENTS_TO_SHOW');
 
-      var isImportantEvent = function(event) {
-        var reason = event.reason;
-        return EVENTS_TO_SHOW_BY_REASON[reason];
+      var isImportantAPIEvent = function(event) {
+        return EVENTS_TO_SHOW_BY_REASON[event.reason];
       };
 
-      var markRead = function(event) {
-        _.set(cachedEvents, [event.metadata.uid, READ], true);
+      var markRead = function(id) {
+        _.set(cachedEvents, [id, READ], true);
         BrowserStore.saveJSON('session','events', cachedEvents);
```

Comment:
If we're saving read and clear status by ID in session storage, we should really change NotificationService to generated a UID instead of using `_.uniqueId`. With `_.uniqueId` the IDs will be reused if you refresh the page, and notification might be incorrectly hidden from the drawer.

Reply:
This is true.  I could pull the `generateSecret` function from `ApplicationGenerator`:

```JavaScript
    scope._generateSecret = function(){
        //http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
        function s4() {
          return Math.floor((1 + Math.random()) * 0x10000)
              .toString(16)
              .substring(1);
          }
        return s4()+s4()+s4()+s4();
      };
```

Reply:
Simpler could just be ` _.uniqueId('notification_') + Date.now() `.